backend/app/services/search_profiles_service.py

```diff
@@ -256,3 +263,27 @@ async def set_search_profile_subscriptions(
             profile_id=request.search_profile_id,
             subscription_ids=request.subscriptions,
         )
+
+    @staticmethod
+    async def get_keyword_suggestions(user: User, suggestions: List[str]) -> KeywordSuggestionResponse:
+        # Avoid useless LLM calls if no topics are available
+        if len(suggestions) == 0:
+            return KeywordSuggestionResponse(suggestions=[])
+
+        prompt = """
+        I will give you a list related keywords. Please add 5 new relevant keywords.\n
```

Comment:
Just an idea, not sure if it makes sense.
```suggestion
        I will give you a list related keywords. Please add 5 new relevant keywords. Don't include synonyms, but suggest words to pin down the topic more exactly with your added relevant keywords. \n
```